backend/src/main/java/org/cryptomator/hub/entities/LegacyAccessToken.java

```diff
@@ -119,4 +133,18 @@ public String toString() {
 					'}';
 		}
 	}
+
+	@ApplicationScoped
+	@Deprecated
+	public static class Repository implements PanacheRepositoryBase<LegacyAccessToken, AccessId> {
+
+		public LegacyAccessToken unlock(UUID vaultId, String deviceId, String userId) {
+			return find("#LegacyAccessToken.get", Map.of("deviceId", deviceId, "vaultId", vaultId, "userId", userId))
+					.firstResult();
```

Comment:
isn't there always a `.singleResult()` method that throws an exception in case of >1 results? we have a primary key on `device_id, vault_id` and don't do any outer joins in this query, so an ambiguous result should be treated as errorneous.

Reply:
fixed in https://github.com/cryptomator/hub/pull/266/commits/6a08760f3c06c6d4a1162287ea74fd16343e1433